src/live-programming.rst

```diff
@@ -47,3 +47,210 @@ TODO: in music
+
+Here is a worry about future technologies, where people interact with technologies mostly with their fingers.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE.bib
+    :filter: key == "bret2011rant"
+
+  The author gives a deep explanation of definition of a tool: A tool addresses human needs by amplifying human capabilities, and a tool convert what people can do to what people want to do. However, most technologies for now focus more on what they can fit people's needs, but less on how they can fit people's capabilities.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE.bib
+    :filter: key == "bret2014humanethought"
+
+  The author holds the opinon that human think not only through part of their body, like visual or auditory, but with the whole parts of body.
+
+In the term programmable physical environment, "environment" is one of the most important part to its definition. Based on the theory of embodied cognition, programmable physical environment should give more focus on interaction between people and the system. For the future design, how people will interact with the environment should be considered more carefully.
+
+Dynamicland
+-----------
+
+Dynamicland is a research group, who are inventing a new computational medium where people work together with real objects in the real world, not alone with virtual objects on screens.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE.bib
+    :filter: key == "dynamicland2019web"
+
+The group have a work that has a close idea to embodied cognition. The computational media is derived from everyday things, like papers, which people can physically feel them and use them. Based on these objects, they are extended with computing capabilities. With the help of cameras, projectors and sensors on the ceiling, each objects can become a computer. The idea is to make a whole room a computer, bring usually virtual things, like programs, into the real world, and build a whold environment where people can think and work physically.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE.bib
+    :filter: key == "steve2018phenomenalworld"
+
+One example about what people can do in dynamicland is as the following.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE.bib
+    :filter: key == "omar2018geokit"
+
+  The `Geokit`, a tool that can build and view map, is not some programs or apps that stored in some virtual places, but some pieces of real paper. With the set of paper, the room get the ability to build and view maps.
+
+Dynamicland actually gives a deep insight on how an environment can be like when combining computer to the whole space. It not only provides a programmable physical system, but a very new way on how people can compute with the environment.
```

Comment:
This just repeats the points that were made above, so you can remove this paragraph.